x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -136,36 +118,10 @@ export class CsvGenerator {
     };
   }
 
-  private getColumns(searchSource: ISearchSource, table: Datatable) {
-    if (this._columns != null) {
-      return this._columns;
-    }
-
-    // if columns is not provided in job params,
-    // default to use fields/fieldsFromSource from the searchSource to get the ordering of columns
-    const getFromSearchSource = (): string[] => {
-      const fieldValues: Pick<SearchSourceFields, 'fields' | 'fieldsFromSource'> = {
-        fields: searchSource.getField('fields'),
-        fieldsFromSource: searchSource.getField('fieldsFromSource'),
-      };
-      const fieldSource = fieldValues.fieldsFromSource ? 'fieldsFromSource' : 'fields';
-      this.logger.debug(`Getting columns from '${fieldSource}' in search source.`);
-
-      const fields = fieldValues[fieldSource];
-      // Check if field name values are string[] and if the fields are user-defined
-      if (isPlainStringArray(fields)) {
-        return fields;
-      }
-
-      // Default to using the table column IDs as the fields
-      const columnIds = table.columns.map((c) => c.id);
-      // Fields in the API response don't come sorted - they need to be sorted client-side
-      columnIds.sort();
-      return columnIds;
-    };
-    this._columns = this.job.columns?.length ? this.job.columns : getFromSearchSource();
-
-    return this._columns;
+  private getColumnsFromTabify(table: Datatable) {
+    const columnIds = table.columns.map((c) => c.id);
```

Comment:
There was room for a lot of cleanup in this file, since we have learned that we can not reliably get the set of column names by checking the fields in the search source. The set of column names simply must be passed separately in the CSV job params: if we try to get the fields from the searchSource we get back info that must be sifted through. That would break the design of this export type, which is intended to NOT be closely bound to implementation details of SearchSource.

Reply:
couldn't we just assume all fields returned by searchsource should be included in the csv export ?( now that discover correctly sets those and we no longer fetch all the fields) ?

Reply:
@ppisljar Discover doesn't always set those fields correctly: https://github.com/elastic/kibana/pull/123412/files/f44648603cc07819509ba8d53262ceb8e048e851#diff-837a43a2ccb9ac8d9f6bf15e96d0e343735c0cd7ad6fdb5d16f6b88dd31b3de1R85

The goal is to keep this part of the code unaware of what did or didn't happen in get_sharing_data. Do you think this will work?

Reply:
i don't know all the details, but i hope that:
- reporting doesn't need to know any details. this report type should be completely generic so every application can consume it. (at the moment only discover does, but i think we want to change that in the future and add csv export support to more applications).
- application requesting a csv report should construct correct search source, which will fetch the minimal information needed to produce the csv (so no columns that are not required in the export)
- we can fix current discover implementation to always construct correct searchsource to pass to reporting, so reporting doesn't need any knowledge about that (as mentioned in the fist bulletpoint)


are some of my assumptions wrong ?

Reply:
@ppisljar Discover and SearchSource seem to have some tech debt that prevents the vision you laid out. But this PR does move Reporting closer to the goal of "reporting doesn't need to know any details." The list of columns is still a necessary piece that Reporting needs to know, due to the afforementioned tech debt.